config/pmd-ruleset.xml

```diff
@@ -64,6 +66,7 @@
     </rule>
 
     <rule ref="category/java/design.xml">
+        <exclude name="AbstractClassWithoutAnyMethod" />
```

Comment:
What's the reason for these extra exclusions? I could go either way on `LiteralsFirstInComparisons` but `UnusedAssignment` and `AbstractClassWithoutAnyMethod` look useful maybe?

Reply:
I updated pmd and those are newly added rules. Maybe they are useful maybe not I just did not investigate it since it is beyond the scope of this pr. I usually take look at such things later.

Reply:
Ah fair enough!